geotrek/trekking/tests/test_parsers.py

```diff
@@ -1148,6 +1150,36 @@ def test_trek_illustration_is_not_imported_on_missing_file_metadata(self, mocked
         self.assertEqual(Attachment.objects.count(), 0)
 
 
+class TestApidaeTrekParserConvertEncodingFiles(TestCase):
+    def test_get_file_with_good_encoding(self):
+        data_dir = "geotrek/trekking/tests/data/"
+        file_name = f'{data_dir}/file_bad_encoding_tmp.kml'
+        copyfile(f'{data_dir}/file_bad_encoding.kml', file_name)
+
+        # Get encoding mode (utf-8, ascii, ISO-8859-1...)
+        detector = UniversalDetector()
```

Comment:
Could you factorize the duplicated code into an inner function? It would improve the readability of your test.

```
def test_get_file_with_good_encoding(self):
  def detect_encoding(filename):
    [...]
    
  # your test here
```
What do you think?